api/v1alpha1/hostedcluster_types.go

```diff
@@ -149,8 +149,25 @@ type ClusterNetworking struct {
 	ServiceCIDR string `json:"serviceCIDR"`
 	PodCIDR     string `json:"podCIDR"`
 	MachineCIDR string `json:"machineCIDR"`
+	// NetworkType specifies the SDN provider used for cluster networking.
+	// +kubebuilder:default:="OpenShiftSDN"
+	// +unionDiscriminator
```

Comment:
what is this unionDiscriminator doing?

Reply:
I believe it's a way to just say "One of the enum types"
But I will get Dan to confirm just because it was a request made in another pr

Reply:
The union tag is only if the value is a struct type representing a union type, e.g.:

```
type NetworkSpec struct { // this is the union type
   Type NetworkType // this is the union discriminator
   SDN *OpenShiftSDNSpec // a possible discriminated value
   Calico *CalicoSpec // a possible discriminated value
}
```

But it looks like this API proposal doesn't allow for there to be any sort of configuration per network type. Which I'm  actually not sure is realistic now that I've said it...

Reply:
Dan clarified I should not be doing this I removed it (thanks for the other good catch)